databuilder/databuilder/extractor/snowflake_usage_extractor.py

```diff
@@ -0,0 +1,106 @@
+    # SELECT statement from snowflake table that contains usage data
+    SQL_STATEMENT = """
+        SELECT
+            database,
+            schema,
+            name,
+            user_email,
+            read_count
+        FROM
+            {database}.{schema}.{table}
+        {where_clause_suffix};
+    """
+    # CONFIG KEYS
+    WHERE_CLAUSE_SUFFIX_KEY = 'where_clause_suffix'
+    POPULARITY_TABLE_DATABASE = 'popularity_table_database'
+    POPULARTIY_TABLE_SCHEMA = 'popularity_table_schema'
+    POPULARITY_TABLE_NAME = 'popularity_table_name'
+
+    DEFAULT_CONFIG = ConfigFactory.from_dict(
+        {WHERE_CLAUSE_SUFFIX_KEY: ' ',
+         POPULARITY_TABLE_DATABASE: 'PROD',
+         POPULARTIY_TABLE_SCHEMA: 'SCHEMA',
+         POPULARITY_TABLE_NAME: 'TABLE'}
+    )
+
+    def init(self, conf: ConfigTree) -> None:
+        conf = conf.with_fallback(SnowflakeUsageExtractor.DEFAULT_CONFIG)
+        self._where_clause_suffix=conf.get_string(SnowflakeUsageExtractor.WHERE_CLAUSE_SUFFIX_KEY)
+        self._popularity_table_database = conf.get_string(SnowflakeUsageExtractor.POPULARITY_TABLE_DATABASE)
+        self._popularity_table_schema = conf.get_string(SnowflakeUsageExtractor.POPULARTIY_TABLE_SCHEMA)
+        self._popularity_table_name = conf.get_string(SnowflakeUsageExtractor.POPULARITY_TABLE_NAME)
+
+        self.sql_stmt = SnowflakeUsageExtractor.SQL_STATEMENT.format(
+            where_clause_suffix=self._where_clause_suffix,
+            database=self._popularity_table_database,
+            schema=self._popularity_table_schema,
+            table=self._popularity_table_name
+        )
+
+        LOGGER.info("SQL for Snowflake popularity: {}".format(self.sql_stmt))
+
+        self._alchemy_extractor = sql_alchemy_extractor.from_surrounding_config(conf, self.sql_stmt)
+        sql_alch_conf = Scoped.get_scoped_conf(conf, self._alchemy_extractor.get_scope()) \
+            .with_fallback(ConfigFactory.from_dict({SQLAlchemyExtractor.EXTRACT_SQL: self.sql_stmt}))
+        self._alchemy_extractor.init(sql_alch_conf)
+        self._extract_iter: Union[None, Iterator] = None
+
+    def extract(self):
+        if not self._extract_iter:
+            self._extract_iter = self._get_extract_iter()
+        try:
+            return next(self._extract_iter)
+        except StopIteration:
+            return None
+
+    def get_scope(self) -> str:
+        return "extractor.snowflake"
+
+    def _get_extract_iter(self):
+        """
+        Using raw level iterator, it groups to table and yields TableColumnUsage
+        :return:
+        """
+        for row in self._get_raw_extract_iter():
+            col_readers = []
+            col_readers.append(ColumnReader(database="snowflake",
+                                            cluster=row["database"],
+                                            schema=row["schema"],
+                                            table=row["name"],
+                                            column="*",
```

Comment:
It looks like `ColumnReader` does currently persist the column in the database (although why, I'm not sure..). Is it worth capturing the column here in the event that `ColumnReader` is updated in the future to do something with that field?

Reply:
Ya I was seeing this: https://github.com/amundsen-io/amundsen/blob/main/databuilder/databuilder/models/table_column_usage.py#L63-L64 so didn't think it'd be worth putting in info about the column because it might be confusing... we could put a note though...

Reply:
yea, makes sense. Column level parsing (such as in your scala script) is also much more complex